### PR TITLE
Revamp EVM logs

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Previously, each EVM log will be emit as a Tendermint log, which is heavy on the Tendermint indexer and also hard to read from. This PR changes it so that EVM logs are stored in application state. Specifically:
- all logs from a transaction is stored inside the transaction receipt
- transaction receipt now also stores log bloom
- if a block has any EVM log, we will also store a block-level bloom, which is the union of all log blooms in that block
During lookup, we will leverage the blooms for faster lookups.

## Testing performed to validate your change
local sei
unit tests

### CODE REVIEW

File: evmrpc/bloom_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	require.Equal(t,
+		"00000000001000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000400000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+		hex.EncodeToString(bloom[:]),
+	)
+	filters := evmrpc.EncodeFilters(
+		[]common.Address{common.HexToAddress("0x797C2dBE5736D0096914Cd1f9A7330403c71d301")},
+		[][]common.Hash{{common.HexToHash("0x036285defb58e7bdfda894dd4f86e1c7c826522ae0755f0017a2155b4c58022e")}},
+	)
+	require.True(t, evmrpc.MatchFilters(bloom, filters))
+}
```
Comment (jewei1997) [inline evmrpc/bloom_test.go:52]: Would be good to test that the bloom match matches with a filter of just the address or just the topics. Also, would be good to test that the bloom does not match with a slightly different address and slightly different filter

File: x/evm/state/log.go [hunk 2]
```diff
@@ -1,53 +1,27 @@
...
-		return
-	}
-	logs.Ls = append(ls, l)
-	logsbz, err := json.Marshal(&logs)
-	if err != nil {
-		s.err = err
-		return
-	}
-	store.Set(LogsKey, logsbz)
+	s.logs = append(s.logs, l)
```
Comment (jewei1997) [inline x/evm/state/log.go:13]: Do we need any concurrency control on these functions?

**Review by jewei1997 — APPROVED** [review]

**Review by jewei1997 — APPROVED** [review]
Overall LGTM

**Review by stevenlanders — APPROVED** [review]

File: evmrpc/bloom_test.go (hunk 1 above)
Comment (codchen) [reply evmrpc/bloom_test.go:52]: yeah good point. Added

File: x/evm/state/log.go (hunk 2 above)
Comment (codchen) [reply x/evm/state/log.go:13]: should be fine since there will be one statedb instance per tx, and there is no parallelization within the tx

**Pull request merged**